### PR TITLE
WOADD-762 :: [GCAT] Import and release max 22-point grades

### PR DESCRIPTION
**Pull request by lee-acn**
*** PLEASE DO NOT OPEN PULL REQUESTS VIA GITHUB ***

The moodle.git repository at Github is just a mirror of the official repository. We do not accept pull requests at Github.

See CONTRIBUTING.txt guidelines for how to contribute patches for Moodle. Thank you.

--

### CODE REVIEW

File: local/gugcat/add/index.php [hunk 1]
```diff
@@ -55,6 +55,9 @@
 $PAGE->set_cm($module);
 
 $scaleid = $module->gradeitem->scaleid;
+if (empty($scaleid) && local_gugcat::is_grademax22($module->gradeitem->gradetype, $module->gradeitem->grademax)){
```
Comment (m-aquino) [inline local/gugcat/add/index.php]: Please use 'is_null()' instead of 'empty()'.

File: local/gugcat/classes/grade_aggregation.php [hunk 2]
```diff
@@ -78,7 +78,11 @@ public static function get_rows($course, $modules, $students){
                 $pg = isset($grades->provisional[$student->id]) ? $grades->provisional[$student->id] : null;
                 $gb = isset($grades->gradebook[$student->id]) ? $grades->gradebook[$student->id] : null;
                 $grd = (isset($pg) && !is_null($pg->finalgrade)) ? $pg->finalgrade : ((isset($gb) && !is_null($gb->grade)) ? $gb->grade : null);
-                local_gugcat::set_grade_scale($item->scaleid);
+                $scaleid = $item->scaleid;
+                if (empty($scaleid) && local_gugcat::is_grademax22($item->gradeitem->gradetype, $item->gradeitem->grademax)){
```
Comment (m-aquino) [inline local/gugcat/classes/grade_aggregation.php]: Please use 'is_null()' instead of 'empty()'.

File: local/gugcat/classes/grade_capture.php [hunk 3]
```diff
@@ -246,7 +257,7 @@ public static function import_from_gradebook($courseid, $module, $students){
                 local_gugcat::update_workflow_state($assign, $student->id, ASSIGN_MARKING_WORKFLOW_STATE_INREVIEW);
```
Comment (m-aquino) [inline local/gugcat/classes/grade_capture.php:257]: Please also add in the $asgrd->grade the '+ $gradescaleoffset'

File: local/gugcat/index.php [hunk 4]
```diff
@@ -60,6 +60,14 @@
     $groups = groups_get_all_groups($course->id, $userid=0, $groupingid, $fields='g.*');
 
     $scaleid = $selectedmodule->gradeitem->scaleid;
+    if (empty($scaleid)){
```
Comment (m-aquino) [inline local/gugcat/index.php]: Please use 'is_null()' instead of 'empty()'.

**Review by m-aquino — CHANGES_REQUESTED** [review]

**Review by m-aquino — APPROVED** [review]

File: local/gugcat/locallib.php [hunk 5]
```diff
@@ -107,6 +109,15 @@ public static function get_grade_grade_items($course, $module){
         return $gradeitems;
     }
 
+    /**
+     * Returns the scale id used for GCAT
+     */
+    public static function get_gcat_scaleid(){
+        global $DB;
+        $gcat_scale = $DB->get_record('scale', array('name' => GCAT_SCALE, 'courseid' => '0'),'id');
+        return $gcat_scale->id;
```
Comment (natividadelijah) [inline local/gugcat/locallib.php]: Please use get_field instead of get_record if you're only going to get 'id'.

**Review by natividadelijah — APPROVED** [review]

**Pull request merged**